### PR TITLE
Update `complete_code_freeze` lane with missing step to push changes

### PR DESCRIPTION
**Pull request by mokagio**
### Description

It's necessary to push when completing the code freeze to ensure that any commit on the branch that is not yet in the remote will reach it so that the beta build will include them.

I adopted the same pattern used by WordPress iOS, see https://github.com/wordpress-mobile/WordPress-iOS/blob/b24d907d0f6c1fffbd5df4347f4f1adefdf82b66/fastlane/lanes/release.rb#L66-L74

### Testing instructions

I tested this by hacking the lane so that 1) it could run on a non-release branch and 2) it would call the release trigger action, but not actually trigger a release (that's a safe test because this PR doesn't change that code)

You can see the changes I made [here](7d7c3878c8656a816227f105bdac60c6bde43d30). The results I got were:

<table>
<tr>
<td>

User input `n`

</td>
<td>

User input `y`

</td>
</tr>
<tr>
<td>

image

</td>
<td>

image

</td>
</tr>
</table>

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -184,8 +184,13 @@ platform :ios do
...
     generate_strings_file_for_glotpress
-    version = ios_get_app_version
-    trigger_beta_build(branch_to_build: "release/#{version}")
+
+    if ENV.fetch('RELEASE_TOOLKIT_SKIP_PUSH_CONFIRM', false) || UI.confirm('Push changes to the remote and trigger the beta build?')
+      push_to_git_remote(tags: false)
+      trigger_beta_build(branch_to_build: "release/#{ios_get_app_version}")
+    else
+      UI.message('Aborting code freeze completion as requested.')
+    end
```
Comment (mokagio) [inline fastlane/Fastfile:193]: As I mentioned in the PR description, this is consistent with what WordPress iOS does.

- Pro: It's good to have a human explicitly push to the remote.
- Con: If you expect the lane to do it's thing without you having to look at it, it might happen that it'll sit there waiting for input for a while, possibly delaying the build going out

⚖️ 

How do you feel about it? 		

<details>
 <summary>My opinion</summary>
 While the "run and forget" option is convenient, as long as we have release steps that need to be initiated by a human on their machine, I think it's appropriate for said human to also pay close attention to what the lane does. As such, I'm leaning towards keeping the blocking request for user input. 
</details>

Comment (AliSoftware) [reply fastlane/Fastfile:193]: Also, reminder that [I've implemented a mechanism a while agao which makes fastlane automatically ping you again on `UI.confirm` and similar interactive prompts if you did not reply after some time](https://github.com/wordpress-mobile/release-toolkit/blob/5c9b79db4bfcb298376fe3e81bc53881795922a5/lib/fastlane/plugin/wpmreleasetoolkit/helper/interactive_prompt_reminder.rb#L3-L22), to remind you that your terminal is waiting for you.

 - By default, your Terminal app should start jumping in your Dock if you didn't reply the interactive prompt after 30s, then 3mn then 10mn — you can adjust those delays using `FASTLANE_PROMPT_REMINDER_DELAYS` env var
 - You can opt-in for this mechanism to also `say` a message after those delays (in addition to making your Dock icon jump) for extra awareness. You can configure and opt into this using the `FASTLANE_PROMPT_REMINDER_MESSAGE` env var

Personally, I use `export FASTLANE_PROMPT_REMINDER_MESSAGE=default` in my `~/.zshrc`, which have saved me a lot of time at a couple of occasions where I left the Terminal after launching the lane and then forgot about it, or if I went getting myself a coffee in the kitchen while the lane was running, and this message notifying me from afar that it was waiting for me 😃  

So TL;DR I'm all ok with that approach. And the fact that you made it possible to opt-out via `RELEASE_TOOLKIT_SKIP_PUSH_CONFIRM` means that anyone who prefers the `git push` to be automattic can still configure their machine to do so, so best of both worlds!

**Review by AliSoftware — APPROVED** [review]

**Review by oguzkocer — APPROVED** [review]
Thanks @mokagio!
